openai/azure-openai/deployment/src/main/java/io/quarkiverse/langchain4j/azure/openai/deployment/AzureOpenAiProcessor.java

```diff
@@ -95,6 +95,7 @@ void generateBeans(AzureOpenAiRecorder recorder,
                 var builder = SyntheticBeanBuildItem
                         .configure(EMBEDDING_MODEL)
                         .setRuntimeInit()
+                        .unremovable()
```

Comment:
How come all these `EmbeddingModel` beans now need to be unremovable?

Reply:
Because potentially only the synthetic generated retrieval augmentor will depend on them, and that AFAIU is not enough to keep them from being removed during build

Reply:
> and that AFAIU is not enough to keep them from being removed during build

Are you sure? If you declare an injection point on the synthetic bean, isn't that enough?

Reply:
Yeah, try removing the `unremovable()` (line 132) in `InProcessEmbeddingProcessor` and run tests of the `easy-rag` extension:

```
java.lang.IllegalStateException: No bean type found for: SYNTHETIC bean [types=[dev.langchain4j.model.embedding.EmbeddingModel, dev.langchain4j.model.embedding.AllMiniLmL6V2QuantizedEmbeddingModel, java.lang.Object], qualifiers=[@Default, @Any], target=n/a]
	at io.quarkus.arc.processor.ComponentsProviderGenerator$BeanAdder.addComponentInternal(ComponentsProviderGenerator.java:695)
	at io.quarkus.arc.processor.ComponentsProviderGenerator$BeanAdder.addComponentInternal(ComponentsProviderGenerator.java:662)
	at io.quarkus.arc.processor.ComponentsProviderGenerator$ComponentAdder.addComponent(ComponentsProviderGenerator.java:811)
	at io.quarkus.arc.processor.ComponentsProviderGenerator.addBeans(ComponentsProviderGenerator.java:417)
	at io.quarkus.arc.processor.ComponentsProviderGenerator.processBeans(ComponentsProviderGenerator.java:250)
	at io.quarkus.arc.processor.ComponentsProviderGenerator.generate(ComponentsProviderGenerator.java:93)
	at io.quarkus.arc.processor.BeanProcessor$1.call(BeanProcessor.java:264)
	at io.quarkus.arc.processor.BeanProcessor$1.call(BeanProcessor.java:260)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	... 5 more

	at io.quarkus.builder.Execution.run(Execution.java:123)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:160)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:330)
	... 3 more
```

Reply:
That seems odd, but it's not terribly important, so let's overlook it for now